packages/flutter/test/painting/image_resolution_test.dart

```diff
@@ -152,6 +150,36 @@ void main() {
         expect(bundleKey.scale, 1.0);
       }));
     });
+
+    test('When the filename contains characters that must be URI-encoded', () {
```

Comment:
Is this test hitting your change in the tool? Or is this verifying something else?

Reply:
Hmm. I think this is an artifact of a different approach I had earlier where I changed the variant lookup rather than the asset manifest itself. Though this test was effective at detecting the bug, I do think this would be more discoverable as a test against `asset.dart`, so I'll get it tested over there instead.